extensions/smallrye-reactive-messaging/runtime/src/main/java/io/quarkus/smallrye/reactivemessaging/runtime/devconsole/DevReactiveMessagingInfos.java

```diff
@@ -138,18 +138,26 @@ public Component getPublisher() {
 
         @Override
         public int compareTo(DevChannelInfo other) {
-            // publisher connectors first
-            if (publisher.type != other.publisher.type) {
-                return publisher.type == ComponentType.CONNECTOR ? -1 : 1;
+            if (publisher != other.publisher) {
+                if (other.publisher == null) {
+                    return -1;
+                }
+                if (publisher == null) {
+                    return 1;
+                }
+                // publisher connectors first
+                if (publisher.type != other.publisher.type) {
+                    return publisher.type == ComponentType.CONNECTOR ? -1 : 1;
+                }
             }
             // consumer connectors last
             long consumerConnetors = consumers.stream().filter(Component::isConnector).count();
             long otherConsumersConnectors = other.consumers.stream().filter(Component::isConnector).count();
             if (consumerConnetors != otherConsumersConnectors) {
                 return Long.compare(otherConsumersConnectors, consumerConnetors);
             }
-
-            if (publisher.type == ComponentType.CONNECTOR && other.publisher.type != ComponentType.CONNECTOR) {
+            if (publisher != other.publisher && publisher.type == ComponentType.CONNECTOR
```

Comment:
won't it NPE if `publisher == other.publisher == null`?

Reply:
Hm, it should not because in that case `publisher != other.publisher` would be false.

Reply:
true, I misread the first check :)